adds create_site_calade() function

djnavarro opened the pull request:
Adds a `create_site_calade()` function that roughly mirrors `create_site_academic()`. 

- It takes an additional `knit` argument because the example site for the calade theme uses .Rmd posts. 
- The example site is the one hosted at https://djnavarro.github.io/hugo-calade/ 
-  Documentation is a little thin at this stage, though there is a brief tutorial on the example site. 

I feel like I'm supposed to write more here but I can't think of anything else! :slightly_smiling_face:

hadley commented on R/site-calade.R:
```diff
@@ -0,0 +1,230 @@
+
+
+# Inserts link to highlight.css file in head_custom.html
+calade_patch_head_custom <- function(path) {
```
Could this be done in calade directly? You'd also need to do it if you were to use with blogdown.

hadley commented on R/site-calade.R:
```diff
@@ -0,0 +1,230 @@
+    '# raw HTML to be passed from the .md file to the .html',
+    '# file. See:',
+    '# https://gohugo.io/getting-started/configuration-markup',
+    '[markup]',
```
Similarly here; I think this is good practice for any hugo site so unless you have strong feelings about it, I think it would be better in calade.

hadley commented on R/site-calade.R:
```diff
@@ -0,0 +1,230 @@
+  inst_version <- rmarkdown::pandoc_version()
+  version <- as.numeric_version(version)
+  if(inst_version < version) {
+    warning("Installation of pandoc is version ", inst_version, ". Calade site may fail to build for pandoc versions below ", version)
```
I think you should make this an error and call the function `check_pandoc()`.

hadley commented on R/site-calade.R:
```diff
@@ -0,0 +1,230 @@
+
+  # stop if no pandoc
+  if(!rmarkdown::pandoc_available()) {
+    stop("Could not find a pandoc installation", call. = FALSE)
```
Can you please use `abort()` here? (It does `call. = FALSE` by default and gives better tracebacks)

hadley commented on R/site-calade.R:
```diff
@@ -0,0 +1,230 @@
+
+  # return early if no pandoc version check is required
+  if(is.null(version)) {
+    return(invisible(NULL))
```
FWIW I usually just write `invisible()` since that seems to work even though I now see it looks like the argument is required.

hadley commented on R/site-calade.R:
```diff
@@ -0,0 +1,230 @@
+
+# Writes the readme file
+calade_write_readme <- function(path) {
+  file_copy(path_package("hugodown", "calade", "README.md"), path)
```
I think you could inline these in `create_site_calade()` since they're so simple.

djnavarro commented:
Hi @hadley - I've moved some of the patches into the calade theme and simplified some of the helper functions. It's throwing these errors on test though. I'm not certain if that's the result of my changes - if it is I'll try to fix it! 

```
 > test_check("hugodown")
── 1. Failure: output gets unicode and colour (@test-md-document.R#40)  ────────
rmd\$lines\[\[10\]\] does not match "#&gt; ✔".
Actual value: ""

-- 2. Failure: interweaving of code and output generates correct html (@test-md-
Results have changed from known value recorded in 'test-md-document-hooks.txt'.
Lengths differ: 60 is not 51
```

hadley commented:
Nope, those are my fault due to some fixes in downlit. If you `pr_merge_main()` you should get the fixes I just pushed into master into your PR.

djnavarro commented:
done!

hadley commented:
And checks passing 🎉  Is this ready for another review pass?

djnavarro commented:
Yes please!

djnavarro commented:
Okay, *now* I'm ready for re-review. No additional changes to the hugodown PR, but I have made a few changes to the calade theme to ensure that when `hugodown::create_site_calade()` is called, the generated site adheres to the "tidy post" format (I think!) and doesn't include unnecessary files.
